core/src/main/java/org/apache/iceberg/io/SortedPosDeleteWriter.java

```diff
@@ -174,20 +208,21 @@ private void flushDeletes() {
     } catch (IOException e) {
       setFailure(e);
       throw new UncheckedIOException(
-          "Failed to write the sorted path/pos pairs to pos-delete file: "
-              + outputFile.encryptingOutputFile().location(),
```

Comment:
Adding the file location would require some confusing code, like:
- Adding a local attribute to the class `lastOutputFile`
- Updating this attribute in `createWriter`
- We need to assume that if there is an error in `FileScopedPositionDeleteWriter` it is always with the last opened writer - which is true, but I am not sure we want to build on that 

I think this is more confusing than what it worth to have the filename in the exception message, but I am open to discussion

Reply:
reverted